aria/orchestrator/workflows/core/events_handler.py

```diff
@@ -147,7 +147,8 @@ def _update_node_state_if_necessary(ctx, is_transitional=False):
     # and also will *never* be the type name
```

Comment:
I know we are currently lacking in the contribution guidelines, but can you please squash all of the three commits into a single commit?